### ISSUE TITLE
Fix error message from cirq.final_wavefunction

### ISSUE THREAD

**Issue by cduck** [quantumlib/Cirq#2778]
If I pass in a non-unitary circuit to `cirq.final_wavefunction` I get an error message with an incorrect suggestion to fix my code.  `cirq.sample_wavefunction` doesn't exist (did it ever?).  It should probably suggest `cirq.final_density_matrix` if the circuit contains channel or mixture operations.

```python
cirq.final_wavefunction(
    cirq.Circuit(
        cirq.DepolarizingChannel(0.1).on(cirq.LineQubit(0))
    )
)
```
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-...> in <module>
----> 1 cirq.final_wavefunction(
      2     cirq.Circuit(
      3         cirq.DepolarizingChannel(0.1).on(cirq.LineQubit(0))
      4     )
      5 )

.../cirq/cirq/sim/mux.py in final_wavefunction(program, initial_state, param_resolver, qubit_order, dtype, seed)
    147     if not protocols.has_unitary(
    148             protocols.resolve_parameters(circuit_like, param_resolver)):
--> 149         raise ValueError(
    150             "Program doesn't have a single well defined final wavefunction "
    151             "because it is not unitary. "

ValueError: Program doesn't have a single well defined final wavefunction because it is not unitary. Maybe you wanted `cirq.sample_wavefunction`?

Program: cirq.Circuit([
    cirq.Moment(operations=[
        cirq.depolarize(p=0.1).on(cirq.LineQubit(0)),
    ]),
])
```

**Comment by jitendrs** [comment]
@vtomole can you please assign it to me.

**Comment by jitendrs** [comment]
Thanks @vtomole

**Comment by jitendrs** [comment]
@vtomole can you please suggest what needs to be done here?

**Comment by vtomole** [comment]
@jitendrs Change https://github.com/quantumlib/Cirq/blob/61b4259db3401afd224dd9eebc83617c93ac341d/cirq/sim/mux.py#L152 to `Maybe you wanted cirq.final_density_matrix?`

**Comment by jitendrs** [comment]
fixed in PR #2844
